Linked Subtree

kodomazer opened the pull request:
Subtree is linked and various updates made.

Not sure if the aliases work for others or if it's local to my machine, but puSys should pull changes from the subtree and shSys should push them to the subtree.

kodomazer committed:
Cleanup

Moved "specialized" entity code into its own example class and made the
internal Entity class abstract.
An Entity is now simply a class that has a statContainer in it, anything
further would be a specialization that is project specific.
Probably want to make Stats and Level (the properties) protected
(instead of public).

Removed all specific StatName references out of the Systems folder.
Hopefully, nothing will break if anyone takes just the Systems folder.
(We should figure out a way to subtree it...)

Note: InventoryManager was also made abstract and one of the  properties
unimplemented (though it probably shouldn't have been implemented as a
base for the system)

kodomazer committed:
Staging

Leaving only systems in place so that we can subtree

kodomazer committed:
Various Changes

Bringing some changes upstream first and then I'll try and subtree

kodomazer committed:
Changes to System

Changed some stuff in systems that wasn't right for some reason.
Lets see if we can push it upstream now.

kodomazer committed:
Revert "Staging"

This reverts commit 57644ac2a7ba0c63d3a83448ff5993f8dc1e8499.

kodomazer committed:
Minor Changes

Just some refactoring.
I'll try updating from the main master and see how it goes from there.

kodomazer committed:
Merge remote-tracking branch 'refs/remotes/Cristian006/master' into update

Updating from the "upstream"

No compiler errors, going to commit and test for functionality before
pushing back to the subtree.

# Conflicts:
#	Assets/Examples/StatSystem/StatSystemTester.cs
#	Assets/Systems/EntitySystem/Entity/Entity.cs
#	Assets/Systems/InventorySystem/InventoryManager.cs

kodomazer committed:
Merge updates into master

Merging into master.
I'll double check that the subtree is up to date and then call it a
night.

# Conflicts:
#	Assets/Systems/StatSystem/Interfaces/IStatLinkable/IStatLinkable.cs

#	Assets/Systems/StatSystem/Interfaces/IStatModifiable/IStatModifiable.cs
#	Assets/Systems/StatSystem/StatCollections/StatCollection.cs
#	Assets/Systems/StatSystem/Stats/Extentions/Attribute/StatAttribute.cs

#	Assets/Systems/StatSystem/Stats/Extentions/Modifiable/StatModifiable.cs

#	Assets/Systems/StatSystem/Stats/Extentions/Regeneratable/StatRegeneratable.cs
#	Assets/Systems/StatSystem/Stats/Extentions/Vital/StatVital.cs
#	Assets/Systems/StatSystem/Stats/Stat.cs

kodomazer committed:
Redoing Subtree

Taking out the directory

kodomazer committed:
Merge commit 'a0296c3fd3ae2b0317e1926d4bbfce6ef633ade9' as 'Assets/Systems'

kodomazer committed:
Squashed 'Assets/Systems/' content from commit 19c7364

git-subtree-dir: Assets/Systems
git-subtree-split: 19c7364c9595b8e2328dccee3b211f2df7318331

kodomazer committed:
Squashed 'Assets/Systems/' changes from 19c7364..fdfa049

fdfa049 Unified everything
26387eb Merge updates into master

git-subtree-dir: Assets/Systems
git-subtree-split: fdfa049b64e1220bf9e06448f02b9ed3c754a901

kodomazer committed:
Merge commit '2a4d5840954a111892b2869e9f7c77a2b1efa781' into subtree

kodomazer committed:
linked subtree

Subtree

Cristian006 commented on Assets/Systems/StatSystem/Stats/Extentions/Vital/StatVital.cs:
```diff
@@ -16,46 +16,57 @@ namespace Systems.StatSystem
-        public int CurrentValue
+        public int Max
+        {
+            get { return base.Value; }
```
I don't think using `base.Value` is necessary here, just stick with `Value`.

kodomazer replied:
base.Value is needed here because I hid the original Value.

Cristian006 commented on Assets/Systems/StatSystem/Stats/Extentions/Vital/StatVital.cs:
```diff
@@ -16,46 +16,57 @@ namespace Systems.StatSystem
             }
         }
+
+        public int Max
```
Why do you have two different `Max` Properties?

Cristian006 commented on Assets/Systems/StatSystem/Stats/Extentions/Vital/StatVital.cs:
```diff
@@ -16,46 +16,57 @@ namespace Systems.StatSystem
+            get { return base.Value; }
+        }
+
+        new public int Value
```
Leave it as `public int Current` or `public int CurrentValue`. Don't hide the base Value property.

kodomazer replied:
I feel like `Value` should represent the current value and not the "max value" that the `Value` of the parent classes represents.  So I decided to hide the parent's `Value` and then show it instead as a `Max` property.

Cristian006 replied:
Okay I understood that and I guess it doesn't really hurt after looking over the test/example code. Maybe just leave `_value` as `_current` or `_currentValue` so there's differentiation between the two.

Cristian006 reviewed (changes requested):
I'll merge it as soon as you fix StatVital. Other than that everything else looks good 😎👌

kodomazer committed:
Renamed and Removed

Renamed _value to _currentValue for readability
Renamed usages of base.Value to Max outside of the definition of
base.Value for redability.

Took out redundant  Max definition.

Cristian006 reviewed (approved):
Yeah this works 👌
